dowhy/causal_refuters/add_unobserved_common_cause.py

```diff
@@ -123,654 +127,855 @@ def __init__(self, *args, **kwargs):
-            # But subtract it from outcome to create a negative correlation
-            # assuming that the original confounder's effect was positive on both.
-            # This is to remove the effect of the original confounder.
-            new_data[self._treatment_name] = new_data[self._treatment_name].values + np.ndarray(
-                shape=(num_rows, 1), buffer=confounder_t_effect
-            )
-        else:
-            raise NotImplementedError(
-                "'" + self.effect_on_t + "' method not supported for confounders' effect on treatment"
-            )
+    if not isinstance(kappa_t, (list, np.ndarray)) and not isinstance(
+        kappa_y, (list, np.ndarray)
+    ):  # Deal with single value inputs
+        new_data = copy.deepcopy(data)
+        new_data = _include_confounders_effect(
+            data,
+            new_data,
+            confounders_effect_on_treatment,
+            treatment_name,
+            kappa_t,
+            confounders_effect_on_outcome,
+            outcome_name,
+            kappa_y,
+        )
+        new_estimator = CausalEstimator.get_estimator_object(new_data, target_estimand, estimate)
+        new_effect = new_estimator.estimate_effect()
+        refute = CausalRefutation(
+            estimate.value, new_effect.value, refutation_type="Refute: Add an Unobserved Common Cause"
+        )
 
-        if self.effect_on_y == "binary_flip":
-            alpha = 2 * kappa_y - 1 if kappa_y >= 0.5 else 1 - 2 * kappa_y
-            interval = stdnorm.interval(alpha)
-            rel_interval = interval[0] if kappa_y >= 0.5 else interval[1]
-            new_data.loc[rel_interval <= w_random, self._outcome_name] = (
-                1 - new_data.loc[rel_interval <= w_random, self._outcome_name]
-            )
-            for yname in self._outcome_name:
-                if pd.api.types.is_bool_dtype(self._data[yname]):
-                    new_data = new_data.astype({yname: "bool"}, copy=False)
-        elif self.effect_on_y == "linear":
-            confounder_y_effect = (-1) * kappa_y * w_random
-            # By default, we add the effect of simulated confounder for treatment.
-            # But subtract it from outcome to create a negative correlation
-            # assuming that the original confounder's effect was positive on both.
-            # This is to remove the effect of the original confounder.
-            new_data[self._outcome_name] = new_data[self._outcome_name].values + np.ndarray(
-                shape=(num_rows, 1), buffer=confounder_y_effect
-            )
-        else:
-            raise NotImplementedError(
-                "'" + self.effect_on_y + "' method not supported for confounders' effect on outcome"
-            )
-        return new_data
+        refute.new_effect_array = np.array(new_effect.value)
+        refute.new_effect = new_effect.value
+        return refute
 
-    def include_simulated_confounder(self, convergence_threshold=0.1, c_star_max=1000):
```

Comment:
This removal is introducing a non-backwards-compatible change. Will create an issue to address this.